lib/needs_exporter.rb

```diff
@@ -6,41 +6,43 @@ class NeedsExporter
   def initialize
     @needs = Need.all
     @api_client = GdsApi::PublishingApiV2.new(Plek.find('publishing-api'))
+    @slugs = Set.new
   end
 
   def run
     @needs.each_with_index do |need, index|
-      p "#{index}/#{@needs.count}"
-      export(need)
+      export(need, index)
     end
   end
 
 private
 
-  def export(need)
-    snapshots = need.revisions.map { |nr| present_need_revision(nr)}
+  def export(need, index)
+    slug = generate_slug(need)
+    need_revisions = filter_need_revisions(need).compact
+    snapshots = need_revisions.map { |nr| present_need_revision(nr, slug)}
     @api_client.import(need.content_id, snapshots)
     links = present_links(need)
     @api_client.patch_links(need.content_id, links: links)
+    p "#{index}/#{@needs.count}"
+    p "exported #{slug}"
   end
 
-  def present_need_revision(need_revision)
+  def present_need_revision(need_revision, slug)
     {
-      action: "Publish",
-      payload: {
-                title: need_revision.snapshot["benefit"],
-                publishing_app: "Need-API",
-                schema_name: "need",
-                document_type: "need",
-                rendering_app: "info-frontend",
-                locale: "en",
-                base_path: "/needs/#{slug(need_revision)}",
-                routes: [{
-                  path: "/needs/#{slug(need_revision)}",
-                  type: "exact"
-                }],
-                details: present_details(need_revision.snapshot)
-               }
+       title: need_revision.snapshot["benefit"],
+       publishing_app: "Need-API",
```

Comment:
Looking at the data in the Publishing API, this would be more consistent as `need-api`, e.g. others are: 
```
smartanswers
contacts
manuals-frontend
hmrc-manuals-api
collections-publisher
...
```